app/functoria_app.ml

```diff
@@ -622,36 +624,64 @@ module Make (P: S) = struct
     with_fmt f
 
   (* Compile the configuration file. *)
-  let compile file =
-    Log.info (fun m -> m "Compiling: %a" Fpath.pp file);
-    let file = Dynlink.adapt_filename (Fpath.basename file)
-    and cfg = Fpath.rem_ext file
-    and config_dir = Fpath.parent !config_file
+  let compile file_ml_path =
+    let build_dir = get_build_dir () in
+    let file = Dynlink.adapt_filename (Fpath.basename file_ml_path)
+    and cfg = Fpath.rem_ext file_ml_path
+    and config_dir = Fpath.(get_cwd () // parent !config_file)
+    and file_ml = Fpath.basename file_ml_path
+    in
+    Log.info (fun m -> m "Compiling: %a" Fpath.pp file_ml_path);
+    let config_dir_to_build_dir = match Fpath.relativize ~root:build_dir config_dir with 
+    | Some x -> x
+    | None -> assert false
     in
-    Bos.OS.Path.matches Fpath.(config_dir / "_build" // cfg + "$(ext)") >>= fun files ->
+    Bos.OS.Path.matches Fpath.(build_dir / "_build" // cfg + "$(ext)") >>= fun files ->
     List.fold_left (fun r p -> r >>= fun () -> Bos.OS.Path.delete p) (R.ok ()) files >>= fun () ->
-    with_current config_dir
+    with_current build_dir
       (fun () ->
-         let pkgs = match P.packages with
-           | []   -> Bos.Cmd.empty
-           | pkgs ->
-             let pkgs =
-               List.fold_left (fun acc pkg ->
-                   String.Set.union pkg.ocamlfind acc
-                 ) String.Set.empty pkgs
-               |> String.Set.elements
-             in
-             Bos.Cmd.(v "-pkgs" % String.concat ~sep:"," pkgs)
-         in
-         let ignore_dirs = match P.ignore_dirs with
-           | []    -> Bos.Cmd.empty
-           | dirs  -> Bos.Cmd.(v "-Xs" % String.concat ~sep:"," dirs)
-         in
-         let cmd =
-           Bos.Cmd.(v "ocamlbuild" % "-use-ocamlfind" % "-classic-display" %
-                    "-tags" % "bin_annot" % "-quiet" %%
-                    ignore_dirs %% pkgs % file)
+          Log.info (fun m -> m "%s" file_ml);
+          let target_dir = Fpath.(v "build-config")
+          (* Import files in the build target *)
+          and target_path = Fpath.(v "build-config" / file_ml)
```

Comment:
what do you think about reusing the `target_dir` variable?

Reply:
No problem with that, so I changed this in last commit.